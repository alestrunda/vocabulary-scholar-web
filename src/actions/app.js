export const APP_INFO_MESSAGE = 'APP_INFO_MESSAGE',
  APP_LOAD_STATE = 'APP_LOAD_STATE',
  APP_SET_ITEMS_PER_PAGE = 'APP_SET_ITEMS_PER_PAGE',
  APP_SET_LANGUAGE = 'APP_SET_LANGUAGE',
  APP_SET_SELECTED_PHRASE = 'APP_SET_SELECTED_PHRASE',
  APP_STATE_NOT_LOADED = 'APP_STATE_NOT_LOADED',
  APP_STORAGE_TYPE = 'APP_STORAGE_TYPE',
  APP_WARNING = 'APP_WARNING',
  APP_WARNING_CLOSE = 'APP_WARNING_CLOSE',
  APP_WORD_LIST_SET_PAGE = 'APP_WORD_LIST_SET_PAGE';

export const appInfoMessage = infoMessage => ({
  type: APP_INFO_MESSAGE,
  payload: { infoMessage },
});

export const appLoadState = (stateToLoad, storageType) => ({
  type: APP_LOAD_STATE,
  payload: { stateToLoad, storageType },
});

export const appSetItemsPerPage = number => ({
  type: APP_SET_ITEMS_PER_PAGE,
  payload: { number },
});

export const appSetLanguage = language => ({
  type: APP_SET_LANGUAGE,
  payload: { language },
});

export const appSetSelectedPhrase = phrase => ({
  type: APP_SET_SELECTED_PHRASE,
  payload: { phrase },
});

export const appStateNotLoaded = () => ({
  type: APP_STATE_NOT_LOADED,
});

export const appStorageType = storageType => ({
  type: APP_STORAGE_TYPE,
  payload: { storageType },
});

export const appWarning = (warning, warningType) => ({
  type: APP_WARNING,
  payload: { warning, warningType },
});

export const appWarningClose = warningType => ({
  type: APP_WARNING_CLOSE,
  payload: { warningType },
});

export const appWordListSetPage = (pageNumber, listID) => ({
  type: APP_WORD_LIST_SET_PAGE,
  payload: { pageNumber, listID },
});
