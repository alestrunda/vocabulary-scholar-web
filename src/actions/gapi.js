export const GAPI_CLIENT_STATUS_UPDATE = 'GAPI_CLIENT_STATUS_UPDATE',
  GAPI_CONNECTED = 'GAPI_CONNECTED',
  GAPI_SET_STORE_FILE_ID = 'GAPI_SET_STORE_FILE_ID',
  GAPI_STORE_LOADED_FROM_DRIVE = 'GAPI_STORE_LOADED_FROM_DRIVE';

export const gapiClientStatusUpdate = (isClientSignedIn, isInitial) => ({
  type: GAPI_CLIENT_STATUS_UPDATE,
  payload: { isClientSignedIn, isInitial },
});

export const gapiConnected = () => ({
  type: GAPI_CONNECTED,
});

export const gapiSetStoreFileID = storeFileID => ({
  type: GAPI_SET_STORE_FILE_ID,
  payload: { storeFileID },
});

export const gapiStoreLoadedFromDrive = (
  isStoreLoadedFromDrive,
  storeFileID
) => ({
  type: GAPI_STORE_LOADED_FROM_DRIVE,
  payload: { isStoreLoadedFromDrive, storeFileID },
});
