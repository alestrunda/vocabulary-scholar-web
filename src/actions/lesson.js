import uniqid from 'uniqid';

export const LESSON_CREATE = 'LESSON_CREATE',
  LESSON_ANONYMOUS_CREATE = 'LESSON_ANONYMOUS_CREATE',
  LESSON_LIST_REMOVE = 'LESSON_LIST_REMOVE',
  LESSON_REMOVE = 'LESSON_REMOVE',
  LESSON_RESET_SCORE = 'LESSON_RESET_SCORE',
  LESSON_SET_SCORE = 'LESSON_SET_SCORE',
  LESSON_VIEW = 'LESSON_VIEW',
  LESSONS_DELETE = 'LESSONS_DELETE',
  LESSONS_IMPORT = 'LESSONS_IMPORT';

export const lessonCreate = ({
  id = uniqid(),
  name = '',
  listIDs = [],
  questionsCnt = 0,
  dateCreated = new Date().getTime(),
  type,
  rateBy,
} = {}) => ({
  type: LESSON_CREATE,
  payload: { id, name, listIDs, questionsCnt, dateCreated, type, rateBy },
});

export const lessonAnonymousCreate = lesson => ({
  type: LESSON_ANONYMOUS_CREATE,
  payload: { lesson },
});

export const lessonListRemove = (lessonID, listID) => ({
  type: LESSON_LIST_REMOVE,
  payload: { lessonID, listID },
});

export const lessonRemove = lessonID => ({
  type: LESSON_REMOVE,
  payload: { lessonID },
});

export const lessonResetScore = lessonID => ({
  type: LESSON_RESET_SCORE,
  payload: { lessonID },
});

export const lessonSetScore = (lessonID, score, finishedIn) => ({
  type: LESSON_SET_SCORE,
  payload: { lessonID, score, finishedIn },
});

export const lessonView = (lessonID, dateViewed = new Date().getTime()) => ({
  type: LESSON_VIEW,
  payload: { lessonID, dateViewed },
});

export const lessonsDelete = () => ({
  type: LESSONS_DELETE,
});

export const lessonsImport = (lessons, dateCreated = new Date().getTime()) => ({
  type: LESSONS_IMPORT,
  payload: { lessons, dateCreated },
});
