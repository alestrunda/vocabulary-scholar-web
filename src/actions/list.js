import uniqid from 'uniqid';

export const LIST_CREATE = 'LIST_CREATE',
  LIST_REMOVE = 'LIST_REMOVE',
  LIST_SET_WORDS = 'LIST_SET_WORDS',
  LIST_SORT = 'LIST_SORT',
  LIST_SORT_RESET = 'LIST_SORT_RESET',
  LIST_VIEW = 'LIST_VIEW',
  LISTS_DELETE = 'LISTS_DELETE',
  LISTS_IMPORT = 'LISTS_IMPORT';

export const listCreate = ({
  id = uniqid(),
  name = '',
  wordIDs = [],
  dateCreated = new Date().getTime(),
} = {}) => ({
  type: LIST_CREATE,
  payload: { id, name, wordIDs, dateCreated },
});

export const listRemove = listID => ({
  type: LIST_REMOVE,
  payload: { listID },
});

export const listSetWords = (listID, selectedWordIDs) => ({
  type: LIST_SET_WORDS,
  payload: { listID, selectedWordIDs },
});

export const listSort = (
  listID,
  sortBy,
  isDescSort,
  words,
  forceResort = false
) => ({
  type: LIST_SORT,
  payload: { forceResort, listID, sortBy, isDescSort, words },
});

export const listSortReset = listID => ({
  type: LIST_SORT_RESET,
  payload: { listID },
});

export const listView = (listID, dateViewed = new Date().getTime()) => ({
  type: LIST_VIEW,
  payload: { listID, dateViewed },
});

export const listsDelete = () => ({
  type: LISTS_DELETE,
});

export const listsImport = (lists, dateCreated = new Date().getTime()) => ({
  type: LISTS_IMPORT,
  payload: { lists, dateCreated },
});
