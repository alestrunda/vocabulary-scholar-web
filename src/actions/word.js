import uniqid from 'uniqid';

export const WORD_ADD_PHRASE = 'WORD_ADD_PHRASE',
  WORD_EDIT_PHRASE = 'WORD_EDIT_PHRASE',
  WORD_LIST_SELECTED = 'WORD_LIST_SELECTED',
  WORD_REMOVE_PHRASE = 'WORD_REMOVE_PHRASE',
  WORD_SET_CUSTOM_FIELD = 'WORD_SET_CUSTOM_FIELD',
  WORD_SET_FAVORITE = 'WORD_SET_FAVORITE',
  WORD_SET_IMAGE_SRC = 'WORD_SET_IMAGE_SRC',
  WORD_SET_AUDIO_SRC = 'WORD_SET_AUDIO_SRC',
  WORD_SET_RATING = 'WORD_SET_RATING',
  WORD_VIEW = 'WORD_VIEW',
  WORDS_DELETE = 'WORDS_DELETE',
  WORDS_IMPORT = 'WORDS_IMPORT';

export const wordAddPhrase = ({
  id = uniqid(),
  wordID,
  phrase,
  startIndex,
  endIndex,
} = {}) => ({
  type: WORD_ADD_PHRASE,
  payload: { id, wordID, phrase, startIndex, endIndex },
});

export const wordEditPhrase = ({
  wordID,
  phraseID,
  phrase,
  startIndex,
  endIndex,
} = {}) => ({
  type: WORD_EDIT_PHRASE,
  payload: { wordID, phraseID, phrase, startIndex, endIndex },
});

export const wordListSelected = (wordID, listID) => ({
  type: WORD_LIST_SELECTED,
  payload: { wordID, listID },
});

export const wordRemovePhrase = (wordID, phraseID) => ({
  type: WORD_REMOVE_PHRASE,
  payload: { wordID, phraseID },
});

export const wordSetCustomField = (wordID, customField) => ({
  type: WORD_SET_CUSTOM_FIELD,
  payload: { wordID, customField },
});

export const wordSetFavorite = ({
  dateAdded = new Date().getTime(),
  wordID,
  word,
  isSettingAsFavorite = false,
} = {}) => ({
  type: WORD_SET_FAVORITE,
  payload: { wordID, dateAdded, word, isSettingAsFavorite },
});

export const wordSetImageSrc = (wordID, imageSrc) => ({
  type: WORD_SET_IMAGE_SRC,
  payload: { wordID, imageSrc },
});

export const wordSetAudioSrc = (wordID, audioSrc) => ({
  type: WORD_SET_AUDIO_SRC,
  payload: { wordID, audioSrc },
});

export const wordSetRating = ({
  wordID,
  rating,
  dateRated = new Date().getTime(),
} = {}) => ({
  type: WORD_SET_RATING,
  payload: { wordID, rating, dateRated },
});

export const wordView = (wordID, dateViewed = new Date().getTime()) => ({
  type: WORD_VIEW,
  payload: { wordID, dateViewed },
});

export const wordsDelete = () => ({
  type: WORDS_DELETE,
});

export const wordsImport = (words, dateAdded = new Date().getTime()) => ({
  type: WORDS_IMPORT,
  payload: { words, dateAdded },
});
