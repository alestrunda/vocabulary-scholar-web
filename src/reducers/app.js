import { ITEMS_PER_PAGE } from '../constants';
import {
  APP_INFO_MESSAGE,
  APP_LOAD_STATE,
  APP_SET_ITEMS_PER_PAGE,
  APP_SET_LANGUAGE,
  APP_SET_SELECTED_PHRASE,
  APP_STATE_NOT_LOADED,
  APP_STORAGE_TYPE,
  APP_WARNING,
  APP_WARNING_CLOSE,
  APP_WORD_LIST_SET_PAGE,
} from '../actions/app';
import { GAPI_CLIENT_STATUS_UPDATE } from '../actions/gapi';

const initState = {
  language: '',
  infoMessage: '',
  phraseSelected: '',
  wordListsActivePages: {},
  itemsPerPage: ITEMS_PER_PAGE,
};

const app = (state = initState, action) => {
  let newState;
  const { payload, type } = action;
  switch (type) {
    case APP_INFO_MESSAGE:
      return {
        ...state,
        infoMessage: payload.infoMessage,
      };
    case APP_LOAD_STATE:
      return {
        ...state,
        ...payload.stateToLoad.app,
        storageType: payload.storageType,
        isStateLoaded: true,
      };
    case APP_SET_ITEMS_PER_PAGE:
      return {
        ...state,
        itemsPerPage: payload.number,
      };
    case APP_SET_LANGUAGE:
      return {
        ...state,
        language: payload.language,
      };
    case APP_SET_SELECTED_PHRASE:
      return {
        ...state,
        phraseSelected: payload.phrase,
      };
    case APP_STATE_NOT_LOADED:
      return {
        ...state,
        isStateLoaded: false,
      };
    case APP_STORAGE_TYPE:
      return {
        ...state,
        storageType: payload.storageType,
      };
    case APP_WARNING:
      if (state[payload.warningType] === false) return state;
      return {
        ...state,
        warning: payload.warning,
        warningType: payload.warningType,
      };
    case APP_WARNING_CLOSE:
      newState = { ...state };
      delete newState.warning;
      newState[payload.warningType] = false;
      return newState;
    case APP_WORD_LIST_SET_PAGE:
      newState = { ...state };
      newState.wordListsActivePages[payload.listID] = payload.pageNumber;
      return newState;
    case GAPI_CLIENT_STATUS_UPDATE:
      if (payload.isInitial) return state;
      return {
        ...state,
        isStateLoaded: false,
        infoMessage: payload.isClientSignedIn
          ? 'Logged in to Google Drive'
          : 'Logged out from Google Drive',
      };
    default:
      return state;
  }
};

export default app;
