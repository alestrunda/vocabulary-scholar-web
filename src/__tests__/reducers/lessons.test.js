import lessonsReducer from '../../reducers/lessons';
import { appLoadState } from '../../actions/app';
import {
  lessonCreate,
  lessonListRemove,
  lessonRemove,
  lessonResetScore,
  lessonSetScore,
  lessonView,
  lessonsDelete,
  lessonsImport,
} from '../../actions/lesson';

it('correct app initial state', () => {
  const action = { type: '' }; //passing no action should return initial state
  const initialState = [];
  const defaultState = undefined;

  expect(lessonsReducer(defaultState, action)).toEqual(initialState);
});

it('does not modify state if action not recognized', () => {
  const action = { type: 'unknown', payload: { someProp: 'some value' } };
  const defaultState = [
    {
      testProp: 'test value',
    },
  ];

  expect(lessonsReducer(defaultState, action)).toEqual(defaultState);
});

describe('APP_LOAD_STATE', () => {
  it('handles APP_LOAD_STATE', () => {
    const stateToLoad = {
        lessons: ['data'],
      },
      storageType = 'local-storage';
    const action = appLoadState(stateToLoad, storageType);
    const defaultState = [];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSON_CREATE', () => {
  it('handles LESSON_CREATE', () => {
    const name = 'lesson',
      listIDs = ['a1', 'b2'],
      questionsCnt = 30,
      type = 'phrase',
      rateBy = 'type-in';
    const action = lessonCreate({
      id: 'dfg564adfg564adf',
      name,
      listIDs,
      questionsCnt,
      dateCreated: 1532979984020,
      type,
      rateBy,
    });
    const defaultState = [];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSON_LIST_REMOVE', () => {
  it('handles LESSON_LIST_REMOVE', () => {
    const lessonID = 'lesson1',
      listID = 'list1';
    const action = lessonListRemove(lessonID, listID);
    const defaultState = [
      {
        id: lessonID,
        name: 'lesson name',
        listIDs: [listID, 'list2', 'list3'],
        questionsCnt: 20,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSON_REMOVE', () => {
  it('handles LESSON_REMOVE', () => {
    const lessonID = 'lesson1';
    const action = lessonRemove(lessonID);
    const defaultState = [
      {
        id: lessonID,
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toEqual([]);
  });
});

describe('LESSON_RESET_SCORE', () => {
  it('handles LESSON_RESET_SCORE', () => {
    const lessonID = 'lesson1';
    const action = lessonResetScore(lessonID);
    const defaultState = [
      {
        id: lessonID,
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
        bestScore: 0.5,
        finishedIn: 150,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSON_SET_SCORE', () => {
  it('handles LESSON_SET_SCORE', () => {
    const lessonID = 'lesson1',
      score = 0.5,
      finishedIn = 150;
    const action = lessonSetScore(lessonID, score, finishedIn);
    const defaultState = [
      {
        id: lessonID,
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSON_VIEW', () => {
  it('handles LESSON_VIEW', () => {
    const lessonID = 'lesson1',
      dateViewed = 1532979984020;
    const action = lessonView(lessonID, 1532979984031);
    const defaultState = [
      {
        id: lessonID,
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
        dateViewed,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});

describe('LESSONS_DELETE', () => {
  it('handles LESSONS_DELETE', () => {
    const action = lessonsDelete();
    const defaultState = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
      {
        id: 'lesson2',
        name: 'lesson 2 name',
        listIDs: ['list3', 'list4', 'list5'],
        questionsCnt: 10,
      },
    ];

    expect(lessonsReducer(defaultState, action)).toEqual([]);
  });
});

describe('LESSONS_IMPORT', () => {
  it('handles LESSONS_IMPORT', () => {
    const defaultState = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
      {
        id: 'lesson2',
        name: 'lesson 2 name',
        listIDs: ['list3', 'list4', 'list5'],
        questionsCnt: 10,
      },
    ];
    const lessonsToImport = [
      {
        id: 'lesson2',
        name: 'lesson 2 name',
        listIDs: ['list3', 'list4', 'list5'],
        questionsCnt: 10,
      },
      {
        id: 'lesson3',
        name: 'lesson 3 name',
        listIDs: ['list10', 'list11', 'list12'],
        questionsCnt: 500,
      },
    ];
    const dateCreated = 1532979984020;
    const action = lessonsImport(lessonsToImport, dateCreated);

    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });

  it("replaces old lesson's props on LESSONS_IMPORT", () => {
    const defaultState = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
    ];
    const lessonsToImport = [
      {
        id: 'lesson1',
        name: 'new lesson name',
        listIDs: ['list3', 'list4'],
        questionsCnt: 25,
      },
    ];
    const dateCreated = 1532979984020;
    const action = lessonsImport(lessonsToImport, dateCreated);
    const result = lessonsReducer(defaultState, action);

    expect(result).toEqual(lessonsToImport);
  });

  it('sets new dateCreated for imported lessons on LESSONS_IMPORT', () => {
    const dateCreated = 1530825000000;
    const defaultState = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
        dateCreated,
      },
    ];
    const lessonsToImport = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
    ];
    const action = lessonsImport(lessonsToImport);

    expect(lessonsReducer(defaultState, action)[0].dateCreated).not.toBe(
      dateCreated
    );
  });

  it('does not affect other lessons on LESSONS_IMPORT', () => {
    const defaultState = [
      {
        id: 'lesson1',
        name: 'lesson name',
        listIDs: ['list1', 'list2', 'list3'],
        questionsCnt: 20,
      },
      {
        id: 'lesson2',
        name: 'lesson 2 name',
        listIDs: ['list3', 'list4', 'list5'],
        questionsCnt: 10,
      },
    ];
    const lessonsToImport = [
      {
        id: 'lesson3',
        name: 'lesson 3 name',
        listIDs: ['list10', 'list11', 'list12'],
        questionsCnt: 500,
        dateCreated: 1530825000000,
      },
    ];
    const action = lessonsImport(lessonsToImport, 1530826000000);
    expect(lessonsReducer(defaultState, action)).toMatchSnapshot();
  });
});
