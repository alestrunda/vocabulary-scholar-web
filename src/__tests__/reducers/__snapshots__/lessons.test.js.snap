// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP_LOAD_STATE handles APP_LOAD_STATE 1`] = `
Array [
  "data",
]
`;

exports[`LESSON_CREATE handles LESSON_CREATE 1`] = `
Array [
  Object {
    "dateCreated": 1532979984020,
    "id": "dfg564adfg564adf",
    "listIDs": Array [
      "a1",
      "b2",
    ],
    "name": "lesson",
    "questionsCnt": 30,
    "rateBy": "type-in",
    "type": "phrase",
  },
]
`;

exports[`LESSON_LIST_REMOVE handles LESSON_LIST_REMOVE 1`] = `
Array [
  Object {
    "id": "lesson1",
    "listIDs": Array [
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
]
`;

exports[`LESSON_RESET_SCORE handles LESSON_RESET_SCORE 1`] = `
Array [
  Object {
    "id": "lesson1",
    "listIDs": Array [
      "list1",
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
]
`;

exports[`LESSON_SET_SCORE handles LESSON_SET_SCORE 1`] = `
Array [
  Object {
    "bestScore": 0.5,
    "finishedIn": 150,
    "id": "lesson1",
    "listIDs": Array [
      "list1",
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
]
`;

exports[`LESSON_VIEW handles LESSON_VIEW 1`] = `
Array [
  Object {
    "dateViewed": 1532979984031,
    "id": "lesson1",
    "listIDs": Array [
      "list1",
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
]
`;

exports[`LESSONS_IMPORT does not affect other lessons on LESSONS_IMPORT 1`] = `
Array [
  Object {
    "id": "lesson1",
    "listIDs": Array [
      "list1",
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
  Object {
    "id": "lesson2",
    "listIDs": Array [
      "list3",
      "list4",
      "list5",
    ],
    "name": "lesson 2 name",
    "questionsCnt": 10,
  },
  Object {
    "dateCreated": 1530826000000,
    "id": "lesson3",
    "listIDs": Array [
      "list10",
      "list11",
      "list12",
    ],
    "name": "lesson 3 name",
    "questionsCnt": 500,
  },
]
`;

exports[`LESSONS_IMPORT handles LESSONS_IMPORT 1`] = `
Array [
  Object {
    "id": "lesson1",
    "listIDs": Array [
      "list1",
      "list2",
      "list3",
    ],
    "name": "lesson name",
    "questionsCnt": 20,
  },
  Object {
    "dateCreated": 1532979984020,
    "id": "lesson2",
    "listIDs": Array [
      "list3",
      "list4",
      "list5",
    ],
    "name": "lesson 2 name",
    "questionsCnt": 10,
  },
  Object {
    "dateCreated": 1532979984020,
    "id": "lesson3",
    "listIDs": Array [
      "list10",
      "list11",
      "list12",
    ],
    "name": "lesson 3 name",
    "questionsCnt": 500,
  },
]
`;
